allennlp/nn/decoding/__init__.py

```diff
@@ -0,0 +1,36 @@
+"""
+This module contains code for transition-based decoding.  If you want to do decoding for a
+vocabulary-based model, where the allowable outputs are the same at every timestep of decoding,
+this code is not what you are looking for, and it will be quite inefficient compared to other
+things you could do.
+
+This code assumes you have a transition-based model, where you start in some state, iteratively
+transition between states, and have some kind of supervision signal that tells you which end
+states, or which transition sequences, are "good".
+
+The key abstractions in this code are the following:
+
+    - ``DecoderState`` represents the current state of decoding, containing a list of all of the
+      actions taken so far, and a current score for the state.  It also has methods around
+      determining whether the state is "finished", for combining states for batched computation.
```

Comment:
What does it mean for a state to be "finished"?  I assume the comma right afterwards should be "and".